classes/local/form/request_evaluation_form.php

```diff
@@ -0,0 +1,305 @@
+        $a->_createElements();
+        $renderer = new \MoodleQuickForm_Renderer();
+        $a->accept($renderer);
+        echo '<div id="evasys-dateselectortemplate" hidden>';
+        echo $renderer->_html;
+        echo '</div>';
+    }
+
+    /**
+     * Defines form elements.
+     */
+    protected function definition() {
+        $mform = $this->_form;
+
+        $mform->addElement('checkbox', 'donotevaluate', get_string('dont_evaluate_course', 'block_evasys_sync'));
+
+        $mform->addElement('html', '<fieldset id="fitem_id_sectionone" class="fitem grouponly" name="sectionone">');
+        $mform->disabledIf('sectionone', 'donotevaluate', 'checked');
+        $mform->disabledIf('sectiontwo', 'donotevaluate', 'checked');
+
+        $mform->addElement('checkbox', 'additionalveranstaltungen',
+            get_string('eval_additional_lsfcourses', 'block_evasys_sync'));
+
+        $mform->addElement('autocomplete', 'lsfcourses',
+            get_string('course_units', 'block_evasys_sync'),
+            $this->lsfcourses, ['multiple' => true]);
+
+        $mform->hideIf('lsfcourses', 'additionalveranstaltungen', 'notchecked');
+
+        $mform->addElement('checkbox', 'additionalcourses',
+            get_string('eval_additional_courses', 'block_evasys_sync'));
+
+        $mform->addElement('course', 'courses', get_string('courses'), ['multiple' => true]);
+        $mform->hideIf('courses', 'additionalcourses', 'notchecked');
+
+        $mform->addElement('html', '</fieldset>'); // Sectionone fieldset.
+
+        $mform->addElement('header', 'evalperiod', get_string('evaluationperiod', 'block_evasys_sync'));
+
+        $mform->addElement('html', '<fieldset id="fitem_id_sectiontwo" class="fitem grouponly" name="sectiontwo">');
+
+        if ($this->defaulttimeframe) {
+            $mform->addElement('checkbox', 'usedefaultevalperiod',
+                get_string('use_default_evalperiod', 'block_evasys_sync',
+                    userdate($this->defaulttimeframe['start']) . ' - ' . userdate($this->defaulttimeframe['end'])));
+            $mform->setDefault('usedefaultevalperiod', 1);
+        }
+
+        $mform->addElement('checkbox', 'useoneevalperiod', get_string('useoneevalperiod', 'block_evasys_sync'));
+        $mform->setDefault('useoneevalperiod', 1);
+
+        $mform->addElement('date_time_selector', 'evaltimestartmulticourse',
+            get_string('startondate', 'block_evasys_sync'));
+        $mform->addElement('date_time_selector', 'evaltimeendmulticourse',
+            get_string('endondate', 'block_evasys_sync'));
+
+        $mform->addElement('date_time_selector', 'evaltimestartsinglecourse',
+            get_string('startondate', 'block_evasys_sync'));
+        $mform->addElement('date_time_selector', 'evaltimeendsinglecourse',
+            get_string('endondate', 'block_evasys_sync'));
+
+        if ($this->defaulttimeframe) {
+            $mform->hideIf('useoneevalperiod', 'usedefaultevalperiod', 'checked');
```

Comment:
I would not completely hide these options, I am sure this will confuse some teachers as they leave the default checkbox and just say there is no option for specifying a custom evaluation period. It would be better to show the options but have them disabled as long as the standard period checkbox is checked.